polkadot-protocol/architecture/system-chains/asset-hub.md

```diff
@@ -1,4 +1,204 @@
+
+Here's a brief overview of the key functions for non-custodial asset transfers:
+
+- **`approve_transfer(asset_id, delegate, amount)`** - approves a delegate to transfer up to a certain amount of the asset on behalf of the original account holder
+- **`cancel_approval(asset_id, delegate)`** - cancels a previous approval for the delegate. Once canceled, the delegate no longer has permission to transfer the approved amount
+- **`transfer_approved(asset_id, owner, recipient, amount)`** - executes the approved asset transfer from the owner’s account to the recipient. The delegate account can call this function once approval is granted
+
+These delegated operations make it easier to manage multi-step transactions and dApps that require complex asset flows between participants.
+
+## Foreign Assets
+
+Foreign assets in Asset Hub refer to assets originating from external blockchains or parachains that are registered in the Asset Hub. These assets are typically native tokens from other parachains within the Polkadot ecosystem or bridged tokens from external blockchains such as Ethereum.
+
+Once a foreign asset is registered in the Asset Hub by its originating blockchain's root origin, users are able to send these tokens to the Asset Hub and interact with them as they would any other asset within the Polkadot ecosystem.
+
+### Handling Foreign Assets
+
+The foreign-assets pallet, an instance of the Assets pallet, manages these assets. Since foreign assets are integrated into the same interface as native assets, developers can use the same functionalities, such as transferring and querying balances. However, there are important distinctions when dealing with foreign assets.
+
+- **Asset identifier** - unlike native assets, foreign assets are identified using an XCM Multilocation rather than a simple numeric `AssetId`. This multilocation identifier represents the cross-chain location of the asset and provides a standardized way to reference it across different parachains and relay chains
+
+- **Transfers** - once registered in the Asset Hub, foreign assets can be transferred between accounts, just like native assets. Users can also send these assets back to their originating blockchain if supported by the relevant cross-chain messaging mechanisms
+
+## Integration
+
+Asset Hub supports a variety of integration tools that make it easy for developers to manage assets and interact with the blockchain in their applications. The tools and libraries provided by Parity Technologies enable streamlined operations, such as querying asset information, building transactions, and monitoring cross-chain asset transfers.
+
+Developers can integrate Asset Hub into their projects using these core tools:
+
+### API Sidecar
```

Comment:
For these tools - is it a good idea to just mention them here, when beyond AssetHub, you could use them for any parachain?

Reply:
As someone new to the eco, this reads to me like API Sidecar is a tool I can use with any Kusama or Polkadot node, so it makes sense to me. I feel the same about the TxWrapper part. I will defer to you and @nhussein11, though. Just let me know if you want me to remove these or do something else.